Sources/StreamChatUI/ChatChannel/StreamModalTransitioningDelegate.swift

```diff
@@ -0,0 +1,166 @@
+/// `UITableView` for the message list component which it doesn't play well with the native modal transition.
+open class StreamModalTransitioningDelegate: NSObject, UIViewControllerTransitioningDelegate {
+    public func presentationController(
+        forPresented presented: UIViewController,
+        presenting: UIViewController?,
+        source: UIViewController
+    ) -> UIPresentationController? {
+        StreamModalPresentationController(presentedViewController: presented, presenting: presenting)
+    }
+}
+
+/// Stream's custom `UIPresentationController` to handle a modal presentation.
+open class StreamModalPresentationController: UIPresentationController {
+    /// The overlay view that is rendered between the presenting and presented view.
+    open var overlayView: UIView = UIView()
+
+    /// The overlay opacity value when the presented view is shown.
+    open var overlayViewAlpha: CGFloat = 0.7
+
+    /// The corner radius of the presented view when it is shown.
+    open var cornerRadius: CGFloat = 22
+
+    /// The presented view height scale.
+    open var presentedViewHeightScale: CGFloat = 0.94
+
+    /// The presenting view height scale when the presented view is shown.
+    open var presentingViewHeightScale: CGFloat = 0.9
+
+    override public init(presentedViewController: UIViewController, presenting presentingViewController: UIViewController?) {
+        super.init(presentedViewController: presentedViewController, presenting: presentingViewController)
+        setup()
+    }
+
+    /// Setup the overlay view and gesture recognizers of the `UIPresentationController`.
+    open func setup() {
+        overlayView.backgroundColor = UIColor.black
+        overlayView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        overlayView.isUserInteractionEnabled = true
+
+        let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(dismissController))
+        overlayView.addGestureRecognizer(tapGestureRecognizer)
+
+        let panGestureRecognizer = UIPanGestureRecognizer(target: self, action: #selector(handlePan))
+        presentedViewController.view.addGestureRecognizer(panGestureRecognizer)
+    }
+
+    override open var frameOfPresentedViewInContainerView: CGRect {
+        guard let containerView = self.containerView else {
+            return .zero
+        }
+        return CGRect(
+            origin: CGPoint(x: 0, y: containerView.frame.height * (1 - presentedViewHeightScale)),
+            size: CGSize(
+                width: containerView.frame.width,
+                height: containerView.frame.height * presentedViewHeightScale
+            )
+        )
+    }
+
+    override open func presentationTransitionWillBegin() {
+        overlayView.alpha = 0
+        containerView?.addSubview(overlayView)
+
+        presentedViewController.transitionCoordinator?.animate(alongsideTransition: { _ in
+            self.showPresentedView()
```

Comment:
Hey @nuno-vieira,
Self is strongly captured here and in other closures. Is this intentional?

Reply:
Yes this is fine in animation closures 👍